### DISCUSSION TITLE
Follow and chat behavior ideas?

### DISCUSSION THREAD

**Discussion by jlplenio** [KevinBytesTheDust/CVAmp#152] [Announcements]
What is the desired behavior for follow and chat functions?
Random messages from a list and follow every x minutes?
Or is the appearance of the user in the chat user list the most important function?

**Comment by yomiyo17** [comment]
It would be nice to get random messages and follows every few minutes actually i think it would be a great feature to have, but i think appearance in the user list takes precedence especially now with the test twitch did and to look more legit.

**Comment by SRZIsHere** [comment]
Personally I think the first one is the most important.

And could be even better if we could choose :
 - The possibility to send a message every X seconds (and not minutes like you have said) 
 - The possibility to put two different timers and the message is send randomly during this time 
 E.g. : Put X = 30s and Y= 2min, the message will be randomly send between 30s and 2 minutes so it'll be less obvious that this is a bot.
 - The possibility to type directly a custom message to send and select which bot you want to send with :
 E.g. : I want the bot number one to say "Hello !", I select 1 + type "Hello !"
 (This function is already available on another bot on github (Kichi779/Twitch-Chat-Bot)
 
 And yes the second option is of course important too !

**Reply by jlplenio** [reply]
We will start with your third option. I have posted an update on ko-fi:
https://ko-fi.com/post/Update-on-the-chat-functionality-progress-C0C2NJLC5

**Comment by jlplenio** [comment]
What do we think about this right mouse button menu mechanic and follow option?

![grafik](https://github.com/jlplenio/crude-twitch-viewer-bot/assets/16845311/1c086126-9b75-4a38-bf9e-5a0a6000664c)

I would love to stick to colors, but the states are getting numerous: 
start, buffering, watching and additionally, not logged in, logged in, no following, following

**Reply by SRZIsHere** [reply]
Perfect !

**Comment by Grillekkj** [comment]
its awesome @jlplenio it would be awesome if u add a "messages.txt" so everybody can just place any message in any language to cvtb send in chat. Just need to be formatted.

**Comment by SRZIsHere** [comment]
Hi !

Just had a feature idea for the future to make the bot less detectable :
- Rotate the "Viewer list"

What I mean :
Actually, when you launch let say 10 Viewers, it will always pick the same 10 First Accounts in the user list (and maybe, I didn't checked it) the 10 first proxies in the proxy list, which means that for Twitch, if they look their logs, they will see that everytime that you launch a stream, it's always the same people in the exact same order that join it. (So it's easily detectable)
 
A way to improve this could be to make a rotation like this :

Everytime the bot is launched, it will assign to an instance a random user from the user list and a random proxy from the proxy list (but this pair user/proxy must always be the same so 1 User will always have the same Proxy IP). 
But it needs to never have the same pair Instance/User+Proxy

E.g. : I launch my stream, launch the bot and send 1 instance to the stream :
This Instance (let say Instance#1) will select a random user/proxy pair (let say the 5th pair that consist of the 5th OAuth token in the userlist which will always be paired with the 5th Proxy in the proxy list.)
If the instance crashes, it will pick again the 5th User+5th Proxy to Instance#1 because it's assigned to it in a variable.

But, If you restart the bot (CTV.exe), this time, like I said it will rotate the list so if you launch 1 Instance, this time :
For Instance#1, it will pick the 10th pair (which will be the 10th user in the userlist and the 10th proxy in the proxylist.)

What do you think ? @jlplenio

**Reply by yomiyo17** [reply]
That's a pretty good idea, i have been doing that by just randomizing my proxie file with a command. It is a nice niche QOL

**Reply by SRZIsHere** [reply]
Is that possible to share the command ?

**Reply by jlplenio** [reply]
I like it. Presently, the proxy list in shuffled at the start once.

Do you want to build semi-permanent pairs of proxy and user? That CTVBot remembers that user X always belongs to proxy Y? That way, each of the users always uses the same proxy/IP? And after a restart, the pairs are maintained, but are used in a different order?

**Reply by SRZIsHere** [reply]
Exactly :)

**Comment by babo963** [comment]
Being in chat and going in raid is the most important thing in my opinion

**Reply by jlplenio** [reply]
Being in chat already works reliably in the latest ko-fi version - let's see if someone can confirm raids over at https://github.com/jlplenio/crude-twitch-viewer-bot/discussions/171#discussioncomment-6538296.

**Reply by babo963** [reply]
i tried the ko-fi version but none of the accounts are shown in chat, first time i tried it, it worked, but next time it didn't

**Reply by SRZIsHere** [reply]
Works perfectly for me, the problem comes from you @babo963

**Reply by babo963** [reply]
Hi @SRZIsHere, how many auth tokens ur using?

**Reply by SRZIsHere** [reply]
~50

**Reply by babo963** [reply]
And they all work for you? or a certain number only?

**Reply by SRZIsHere** [reply]
All of them, maybe you did not get the oauths correctly, are they detected ?

**Reply by babo963** [reply]
What do u mean by are they detected? how can i check that

**Reply by babo963** [reply]
@SRZIsHere Some of them worked after changing the password which generated a new auth token, How do i check if they are detected?
